compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -5260,6 +5260,111 @@ static bool skipRemLowering(int64_t divisor, TR::Simplifier *s)
    }
 
 
+/* Reduce Ifcmp Greater-Than-Or-Equal to Equal in the following case. If
+* - the first child is an "and" node
+* - the second child of the "and" node is a constant N
+* - the second child of the ifcmple node is a constant N,
+* Then, we can reduce ifcmple to ifcmpeq.
+*
+* For example, convert
+* ifiucmple --> block_5
+*  iand
+*    iload  <...>
+*    iconst 16
+*  iconst 16
+*
+*  into
+*
+* ifiucmpeq --> block_5
+*  iand
+*    iload  <...>
+*    iconst 16
+*  iconst 16
+*
+* Since the constants are equal, the case of greater-than/less-than is impossible,
+* and we can reduce a GE/LE operation to an EQ operation. After the reduction is performed,
+* the test-under-mask optimization can be applied to the "ifcmpeq" node.
+*/
```

Comment:
Yes, X86 support memory vs constant comparison. By reducing  `ifcmpge/le` to `ifcmpeq`, X86 could potentially generate one instruction less. It can never be worse (in theory).